File: String/1374. Generate a String With Characters That Have Odd Counts.py
class Solution:
    def generateTheString(self, n: int) -> str:
        if not n:
            return ''
        if n == 1:
            return 'a'
        if n % 2 == 0:
            return 'a'*(n-1) + 'b'
        return 'a'*n
        
class Solution:
    def generateTheString(self, n):
        return 'b' + 'ab'[n & 1] * (n - 1)

File: String/1374. Generate a String With Characters That Have Odd Counts.cpp
class Solution {
public:
    string generateTheString(int n) {
        return "b" + string(n - 1, 'a' + n % 2);
    }
};

class Solution {
public:
    string generateTheString(int n) {
        return n % 2 ? string(n, 'a') : string(n - 1, 'a') + 'b';
    }
};